src/NUnitTestExtractor/NUnitTestExtractor.Tests/NUnitTestExtractorTests.cs

```diff
@@ -3,22 +3,309 @@
+
+            if (directory != null && !Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+            }
+
+            if (!File.Exists(outputDirectory))
+            {
+                using (FileStream stream = File.Create(outputDirectory)) { }
+            }
+        } 
+
+        [TearDown]
+        public void EraseOutputFile()
+        {
+            if (File.Exists(outputDirectory))
+            {
+                File.Delete(outputDirectory);
+            }
+        }
+
+        [TestCaseSource(typeof(InvalidDataTestCases))]
+        public void WritingToFile_InvalidDll_NothingWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            using (StreamReader reader = new StreamReader(outputDirectory))
+            {
+                string lines = reader.ReadToEnd();
+
+                Assert.That(lines, Is.Empty);
+            }
+        }
+
+        [TestCaseSource(typeof(ValidDataTestCases))]
+        public void WritingToFile_ValidDllAndEveryLevel_GetsWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            using (StreamReader reader = new StreamReader(outputDirectory))
+            {
+                string lines = reader.ReadToEnd();
+
+                Assert.That(lines, Is.Not.Empty);
+            }
+        }
+
+        [TestCase("C:/Git/NTestController/bin/Debug/NTestControllerTests.dll", NUnitTestExtractorApp.Level.Namespace)]
+       public void WritingToFile_LevelIsNamespace_OnlyNamespaceWritten(string dll, NUnitTestExtractorApp.Level level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, level);
+
+            string[] lines = File.ReadAllLines(outputDirectory);
+            string result = lines[0].Substring(lines[0].IndexOf("|")+2);
+
+            bool containsPeriod = result.Contains(".");
+
+            Assert.That(!containsPeriod);
+        }
+
+        [TestCase("C:/Git/NTestController/bin/Debug/NTestControllerTests.dll", NUnitTestExtractorApp.Level.Class)]
+        public void WritingToFile_LevelIsClass_NamespaceAndClassWritten(string dll, NUnitTestExtractorApp.Level level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, level);
+
+            string[] lines = File.ReadAllLines(outputDirectory);
+            string result = lines[0].Substring(lines[0].IndexOf("|") + 2);
             
+            //This works as class and namespace are seperated by a period example: namepspace.class
+            Assert.That(result.Contains("."));
```

Comment:
But it should verify that only 1 "." was found, and that there's 1 or more characters before & after the "."
You could use Regex to verify the format, and that would allow you to also validate only alphanumeric or _ characters in the namespace & class.
In fact, you could even use a Regex string to validate the entire line (DLL | namespace.class)...